src/panels/flow-histogram-react/FlowHistogramHelpers.ts

```diff
@@ -1,96 +1,217 @@
+                break
+            case 'Conversations':
+                metricLabels = getSeriesMetricValues(sd.fields, 'Application')
+                const sourceLabels = getSeriesMetricValues(sd.fields, 'Source')
+                const destLabels = getSeriesMetricValues(sd.fields, 'Dest.')
+                labelFunction = (name: string, idx: number): string => {
+                    return sourceLabels[idx] + ' <-> ' + destLabels[idx] + ' [' + (name ? name : 'Unknown') + ']'
+                }
+                break
+        }
+
+        const indexedMetricLabels = getIndexedMetricLabels(metricLabels, labelFunction, stacked)
+        const timeRangeInSeconds = moment.duration(data.timeRange.to.diff(data.timeRange.from)).asSeconds();
+
+        if (!stacked) {
+            inResult = getSeparatedResultData(inByData, indexedMetricLabels, horizontal, unitInfo, rate, timeRangeInSeconds)
+            outResult = getSeparatedResultData(outByData, indexedMetricLabels, horizontal, unitInfo, rate, timeRangeInSeconds)
+
+        } else {
+            stackedResult = getStackedResultData(indexedMetricLabels, inByData, outByData, horizontal, unitInfo, rate, timeRangeInSeconds)
+        }
+
+        labeledValues = {
+            separateData: {
+                inByLabel: inResult,
+                outByLabel: outResult,
+                ticks: indexedMetricLabels
+            },
+            stackedData: {
+                metricsByLabel: stackedResult,
+                ticks: [[FLowDataDirection.dataIn.value, FLowDataDirection.dataIn.label], [FLowDataDirection.dataOut.value, FLowDataDirection.dataOut.label]]
+            }
+        }
+        return labeledValues
+
+    } else {
+        throw new Error('Only one query is permited in this panel');
+    }
+    return labeledValues
+}
+
+/**
+ * Returns and array of [index, label] 
+ * where index is the record position in the original data and the label to be displayed in the graph
+ * @param metricLabels 
+ * @param labelFunction 
+ * @param stacked 
+ * @returns 
+ */
+export const getIndexedMetricLabels = (metricLabels: any[], labelFunction, stacked: boolean) => {
+    return metricLabels.map((label, idx) => [idx, labelFunction(label, idx)])
+}
+
+/**
+ * Gets the final data formatted to be passed into the plot function 
+ * based in graph direction (vertical, horizontal) and mode (separated or stacked)
+ * @param data 
+ * @param indexedLabels 
+ * @param horizontal 
+ * @param stacked 
+ * @returns 
+ */
+export const getSeparatedResultData = (data: any[], indexedLabels: any[], horizontal: boolean, unitInfo: FlowPanelUnitInfo, rate: boolean, timeRangeInSeconds: number) => {
+    const divisor = rate ? unitInfo.divisor * timeRangeInSeconds : unitInfo.divisor
+    return indexedLabels.map((il) => {
+        const index = il[DataPosition.indexLabel.index]
+        const value = data[index] / divisor
+        return horizontal ? [value, index] : [index, value]
+    })
+}
+
```

Comment:
Nit: I think a comment/documentation would be useful here. These helpers are usually out of context of where they're used, so the more context you can bring via comments will be useful in the future.

Reply:
Will do!, this is a flot thing that when displaying horizontal or vertical bars the data needs to switch